Sources/TokamakCore/Shapes/ModifiedShapes.swift

```diff
@@ -52,6 +62,20 @@ public struct _TrimmedShape<S>: Shape where S: Shape {
       .path(in: rect)
       .trimmedPath(from: startFraction, to: endFraction)
   }
+
+  public typealias AnimatableData = AnimatablePair<
+    S.AnimatableData,
+    AnimatablePair<CGFloat, CGFloat>
+  >
+  public var animatableData: AnimatableData {
+    get {
+      .init(shape.animatableData, .init(startFraction, endFraction))
+    }
+    set {
+      shape.animatableData = newValue[].0
```

Comment:
Huh, I didn't know that empty subscripts are possible? Would you mind clarifying how does it actually work?

Reply:
The empty subscript just returns a tuple of the `(first, second)` elements of the pair. SwiftUI has it internally, but it'd work just as well as a property.